.github/workflows/ci.yml

```diff
@@ -40,6 +36,10 @@ jobs:
         node-version: ${{ matrix.node-version }}
         cache: 'npm'
 
+    - name: Update npm
+      if: matrix.node-version == '6.x'
+      run: npm install -g npm@^5
```

Comment:
node 6.x installs npm@3 which had issues installing all dependencies of jest. Updating to npm@5 fixes the issue, and didn't think it worth spending time trying to figure out why it wasn't working.